### PR TITLE
Remove HeapSizeOf-bound for Hasher::Out and add it only where needed

### PR DESCRIPTION
**Pull request by dvdplm**
The `HeapSizeOf` bound is causing trouble when [integrating `parity-common` with polkadot](https://github.com/paritytech/polkadot/pull/297) and is probably best added only where needed.

### CODE REVIEW

**Comment by tomaka** [comment]
More generally, `heapsize` has been determined by their author to be a bad idea and should eventually be removed from all our code bases.

**Comment by dvdplm** [comment]
@tomaka I for one is hating it with a vengeance. So, is that a 👍🏿 from you?

File: plain_hasher/util/plain_hasher/Cargo.toml [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-version = "0.1.0"
-authors = ["Parity Technologies <admin@parity.io>"]
-license = "MIT"
-keywords = ["hash", "hasher"]
-homepage = "https://github.com/paritytech/plain_hasher"
-
-[dependencies]
-crunchy = "0.1.6"
-ethereum-types = "0.3"
-hashdb = { version = "0.2.0", path = "../hashdb" }
```
Comment (debris) [inline plain_hasher/util/plain_hasher/Cargo.toml]: what happened with this file? ;)

Comment (dvdplm) [reply plain_hasher/util/plain_hasher/Cargo.toml]: It was a left-over from when I moved over the code from `parity-ethereum` – I must have messed up the `git subtree` command at some point.

**Pull request merged**